cv_analysis/cv_submission/views.py

```diff
@@ -0,0 +1,255 @@
+from django.shortcuts import render
+from rest_framework.generics import GenericAPIView
+from job_description.serializers import JDSerializer
+from rest_framework import response, status, permissions
+from .models import JD_CV
+from job_description.models import JD
+from cv.models import CURRICULUMVITAE
+from cv_submission.serializers import JD_CVSerializer
+from django.core.exceptions import ObjectDoesNotExist
+from authentication.permission import IsHR, IsCandidate
+from authentication.models import User
+from cv_analysis.services.compare_CV_JD import GeminiCompareService
+
+# Create your views here.
+class JD_CVAPIView(GenericAPIView):
+    serializer_class = JD_CVSerializer
+
+    def get_permissions(self):
+        if self.request.method == 'GET':
+            self.permission_classes = [IsHR]
+        elif self.request.method == 'POST':
+            self.permission_classes = [IsCandidate]
+        else:
+            self.permission_classes = [permissions.AllowAny]
+        return super(JD_CVAPIView, self).get_permissions()
+    
+    def get(self, request, id):
+        recommendation = request.query_params.get('recommendation', None)
+
+        try:
+            jd = JD.objects.get(id=id)
+        except JD.DoesNotExist:
+            return response.Response({'message': 'JD not found'}, status=status.HTTP_404_NOT_FOUND)
+
+        jd_cv_queryset = JD_CV.objects.filter(jd_id=id)
+        
+        if recommendation:
+            filtered_jd_cv = []
+            for jd_cv_item in jd_cv_queryset:
+                if (jd_cv_item.cv_analysis and 
+                    isinstance(jd_cv_item.cv_analysis, dict) and
+                    jd_cv_item.cv_analysis.get('recommendation') == recommendation):
+                    filtered_jd_cv.append(jd_cv_item)
+            
+            jd_cv = filtered_jd_cv
+            total_submissions = len(filtered_jd_cv)
+        else:
+            jd_cv = jd_cv_queryset
+            total_submissions = jd_cv_queryset.count()
```

Comment:
Use not recommendation to avoid nested if block. You can refer to the code below
```suggestion
        if not recommendation:
            jd_cv = jd_cv_queryset
            total_submissions = jd_cv_queryset.count()
            
        filtered_jd_cv = []
        for jd_cv_item in jd_cv_queryset:
            if (jd_cv_item.cv_analysis and 
                isinstance(jd_cv_item.cv_analysis, dict) and
                jd_cv_item.cv_analysis.get('recommendation') == recommendation):
                filtered_jd_cv.append(jd_cv_item)
            
        jd_cv = filtered_jd_cv
        total_submissions = len(filtered_jd_cv) 
```